packages/fast-usdc/src/exos/settler.js

```diff
@@ -1,97 +1,257 @@
+      tap: M.interface('SettlerTapI', {
+        receiveUpcall: M.call(M.record()).returns(M.promise()),
+      }),
+      notify: M.interface('SettlerNotifyI', {
+        notifyAdvancingResult: M.call(
+          M.string(),
+          M.nat(),
+          M.boolean(),
+        ).returns(),
+      }),
+      self: M.interface('SettlerSelfI', {
+        disburse: M.call(EvmHashShape, M.string(), M.nat()).returns(
+          M.promise(),
+        ),
+        forward: M.call(
+          M.opt(EvmHashShape),
+          M.string(),
+          M.nat(),
+          M.string(),
+        ).returns(M.promise()),
+      }),
+    },
     /**
-     *
      * @param {{
      *   sourceChannel: IBCChannelID;
-     *   remoteDenom: Denom
+     *   remoteDenom: Denom;
+     *   repayer: LiquidityPoolKit['repayer'];
+     *   settlementAccount: HostInterface<OrchestrationAccount<{ chainId: 'agoric' }>>
      * }} config
      */
-    config => harden(config),
+    config => {
+      return {
+        ...config,
+        /** @type {HostInterface<TargetRegistration>|undefined} */
+        registration: undefined,
+        /** @type {SetStore<ReturnType<typeof makeMintedEarlyKey>>} */
+        mintedEarly: zone.detached().setStore('mintedEarly'),
+      };
+    },
     {
-      /** @param {VTransferIBCEvent} event */
-      async receiveUpcall(event) {
-        if (event.packet.source_channel !== this.state.sourceChannel) {
-          // TODO #10390 log all early returns
-          // only interested in packets from the issuing chain
-          return;
-        }
-        const tx = /** @type {FungibleTokenPacketData} */ (
-          JSON.parse(atob(event.packet.data))
-        );
-        if (tx.denom !== this.state.remoteDenom) {
-          // only interested in uusdc
-          return;
-        }
-
-        if (!addressTools.hasQueryParams(tx.receiver)) {
-          // only interested in receivers with query params
-          return;
-        }
-
-        const { EUD } = addressTools.getQueryParams(tx.receiver);
-        if (!EUD) {
-          // only interested in receivers with EUD parameter
-          return;
-        }
-
-        // TODO discern between SETTLED and OBSERVED; each has different fees/destinations
-        const hasPendingSettlement = statusManager.hasPendingSettlement(
+      creator: {
+        async monitorMintingDeposits() {
+          const { settlementAccount } = this.state;
+          const registration = await vowTools.when(
+            settlementAccount.monitorTransfers(this.facets.tap),
+          );
+          assert.typeof(registration, 'object');
+          this.state.registration = registration;
+        },
+      },
+      tap: {
+        /** @param {VTransferIBCEvent} event */
+        async receiveUpcall(event) {
+          if (event.packet.source_channel !== this.state.sourceChannel) {
+            // TODO #10390 log all early returns
+            // only interested in packets from the issuing chain
+            return;
```

Comment:
Before closing #10390, please log all early returns. This was a previous request: https://github.com/Agoric/agoric-sdk/pull/10406/files#r1831815520